django/http/multipartparser.py

```diff
@@ -166,15 +175,32 @@ def parse(self):
                 field_name = force_text(field_name, encoding, errors='replace')
 
                 if item_type == FIELD:
+                    # Avoid storing more than DATA_UPLOAD_MAX_NUMBER_FIELDS
+                    num_post_keys += 1
+                    if (settings.DATA_UPLOAD_MAX_NUMBER_FIELDS is not None and
+                            settings.DATA_UPLOAD_MAX_NUMBER_FIELDS < num_post_keys):
+                        raise TooManyFieldsSent('Too many fields sent. Check DATA_UPLOAD_MAX_NUMBER_FIELDS.')
+
+                    # Avoid reading more than DATA_UPLOAD_MAX_MEMORY_SIZE
+                    if settings.DATA_UPLOAD_MAX_MEMORY_SIZE is not None:
+                        read_size = settings.DATA_UPLOAD_MAX_MEMORY_SIZE - self._data_size
```

Comment:
```
read_size = settings.DATA_UPLOAD_MAX_MEMORY_SIZE - self._data_size
```
could become:
```
read_size = settings.DATA_UPLOAD_MAX_MEMORY_SIZE - self._data_size - len(field_name)
```
because the field name has already been read into memory at that point. 

In any case, it seems that the code may stay the way it is since the `field_name` is limited (I believe) to some reasonable size.